internal/secrets/encryptor_test.go

```
@@ -0,0 +1,285 @@
+		// Only load secondary key to encrypt
+		e := newEncryptor(secondaryKey, nil)
+		encrypted, err := e.EncryptBytes(messageToEncrypt)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Then load both keys to decrypt
+		e = newEncryptor(primaryKey, secondaryKey)
+		decrypted, err := e.DecryptBytes(encrypted)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if !bytes.Equal(decrypted, messageToEncrypt) {
+			t.Fatal("unable to decrypt and get the original bytes")
+		}
+	})
+}
+
+func TestNoOpEncryptor(t *testing.T) {
+	e := noOpEncryptor{}
+
+	encrypted, err := e.EncryptBytes(messageToEncrypt)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !bytes.Equal(encrypted, messageToEncrypt) {
+		t.Fatal("encrypted bytes is not same as the original")
+	}
+}
+
+func TestBadKeysFailToDecrypt(t *testing.T) {
+	key, err := generateRandomAESKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	e := newEncryptor(key, nil)
+
+	encrypted, err := e.EncryptBytes(messageToEncrypt)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	decrypted, err := e.DecryptBytes(encrypted)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	notTheSameKey, err := generateRandomAESKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+	e = newEncryptor(notTheSameKey, nil)
+
+	decryptedAgain, err := e.DecryptBytes(encrypted)
+	if err == nil {
+		t.Fatal("Should not have been able to Decrypt string with an invalid secret key.")
+	}
+
+	if bytes.Equal(decrypted, decryptedAgain) {
+		t.Fatal("Should not have been able to Decrypt string with an invalid secret key.")
+	}
+}
+
+// Test that different strings EncryptBytes to different outputs
+func TestDifferentOutputs(t *testing.T) {
+	key, _ := generateRandomAESKey()
+	e := newEncryptor(key, nil)
+	messages := []string{
+		"This may or may",
+		"This is not the same as that",
+		"The end of that",
+		"Plants and animals",
+		"Snow, igloos, sunshine, unicords",
+	}
+
+	var crypts []string
+	for _, m := range messages {
+		encrypted, _ := e.EncryptBytes([]byte(m))
```

Check the error or add a code comment to explain why it is OK to ignore the error here in particular.